test/e2e/cadvisor.go

```diff
@@ -44,14 +50,15 @@ func CheckCadvisorHealthOnAllNodes(c *client.Client, timeout time.Duration) {
 	nodeList, err := c.Nodes().List()
 	expectNoError(err)
 	var errors []error
-	for start := time.Now(); time.Since(start) < timeout; time.Sleep(5 * time.Second) {
+	retries := maxRetries
+	for {
 		errors = []error{}
 		for _, node := range nodeList.Items {
 			// cadvisor is not accessible directly unless its port (4194 by default) is exposed.
 			// Here, we access '/stats/' REST endpoint on the kubelet which polls cadvisor internally.
 			statsResource := fmt.Sprintf("api/v1beta1/proxy/minions/%s/stats/", node.Name)
 			By(fmt.Sprintf("Querying stats from node %s using url %s", node.Name, statsResource))
-			_, err = c.Get().AbsPath(statsResource).Timeout(1 * time.Second).Do().Raw()
+			_, err = c.Get().AbsPath(statsResource).Timeout(timeout).Do().Raw()
```

Comment:
Am I reading this right? Each timeout is a minute, across 2 nodes, for 10 retries, so you want a budget of 20 minutes for this test to fail?